docs/src/modules/components/Demo.js

```diff
@@ -196,16 +197,101 @@ class Demo extends React.Component {
     this.handleCloseMore();
   };
 
+  handleCodeLanguageClick = event => {
+    const codeVariant = event.currentTarget.value;
+
+    if (this.props.options.codeVariant !== codeVariant) {
+      document.cookie = `codeVariant=${codeVariant};path=/;max-age=31536000`;
+
+      this.props.dispatch({
+        type: ACTION_TYPES.OPTIONS_CHANGE,
+        payload: {
+          codeVariant,
+        },
+      });
+    }
+
+    this.setState(prevState => {
+      return {
+        /**
+         * if the the same code type is open,
+         * toggle the state, otherwise if it is
+         * another code type always open it. i.e, true
+         */
+        codeOpen: this.props.options.codeVariant === codeVariant ? !prevState.codeOpen : true,
+      };
+    });
+  };
+
+  handleClickCodeOpen = () => {
+    this.setState(state => ({
+      codeOpen: !state.codeOpen,
+    }));
+  };
+
+  getDemoData = () => {
+    const { options, demo } = this.props;
+    return options.codeVariant === CODE_VARIANTS.HOOK && demo.rawHook
+      ? {
+          codeVariant: CODE_VARIANTS.HOOK,
+          raw: demo.rawHook,
+          js: demo.jsHook,
+        }
+      : {
+          codeVariant: CODE_VARIANTS.JS,
+          js: demo.js,
+          raw: demo.raw,
+        };
+  };
+
   render() {
-    const { classes, demoOptions, githubLocation, index, js: DemoComponent, raw } = this.props;
+    const { classes, demo, demoOptions, githubLocation: githubLocationJS } = this.props;
     const { anchorEl, codeOpen } = this.state;
     const category = demoOptions.demo;
+    const demoData = this.getDemoData();
+    const DemoComponent = demoData.js;
+    const githubLocation =
+      demoData.codeVariant === CODE_VARIANTS.HOOK
+        ? githubLocationJS.replace(/\.jsx?$/, '.hook.js')
```

Comment:
@oliviertassinari  Should I update my PR https://github.com/mui-org/material-ui/pull/13497 to accommodate `hook.js` instead of `hooks.js`?

Reply:
It's a very good question, using hook over hooks is only a suggestion. Given a demo will most likely host multiple hooks, I think that we can revert my change 👍.